components/org.wso2.carbon.aws/src/main/java/org/wso2/carbon/aws/user/store/mgt/util/AWSSignatureV4Generator.java

```diff
@@ -0,0 +1,421 @@
+        canonicalURI = builder.canonicalURI;
+        queryParametes = builder.queryParametes;
+        awsHeaders = builder.awsHeaders;
+        payload = builder.payload;
+
+        /* Get current timestamp value.(UTC) */
+        xAmzDate = getTimeStamp();
+        currentDate = getDate();
+    }
+
+    /**
+     * Task 1: Create a Canonical Request for Signature Version 4.
+     *
+     * @return Canonical Request.
+     */
+    private String prepareCanonicalRequest() {
+
+        StringBuilder canonicalURL = new StringBuilder("");
+
+        /* Step 1.1 Start with the HTTP request method (GET, PUT, POST, etc.), followed by a newline character. */
+        canonicalURL.append(httpMethodName).append("\n");
+
+        /* Step 1.2 Add the canonical URI parameter, followed by a newline character. */
+        canonicalURI = canonicalURI == null || canonicalURI.trim().isEmpty() ? "/" : canonicalURI;
+        canonicalURL.append(canonicalURI).append("\n");
+
+        /* Step 1.3 Add the canonical query string, followed by a newline character. */
+        StringBuilder queryString = new StringBuilder("");
+        if (queryParametes != null && !queryParametes.isEmpty()) {
+            for (Map.Entry<String, String> entrySet : queryParametes.entrySet()) {
+                String key = entrySet.getKey();
+                String value = entrySet.getValue();
+                queryString.append(key).append("=").append(encodeParameter(value)).append("&");
+            }
+            queryString.deleteCharAt(queryString.lastIndexOf("&"));
+            queryString.append("\n");
+        } else {
+            queryString.append("\n");
+        }
+        canonicalURL.append(queryString);
+
+        /* Step 1.4 Add the canonical headers, followed by a newline character. */
+        StringBuilder signedHeaders = new StringBuilder("");
+        if (awsHeaders != null && !awsHeaders.isEmpty()) {
+            for (Map.Entry<String, String> entrySet : awsHeaders.entrySet()) {
+                String key = entrySet.getKey();
+                String value = entrySet.getValue();
+                signedHeaders.append(key).append(";");
+                canonicalURL.append(key).append(":").append(value).append("\n");
+            }
+
+            /* Note: Each individual header is followed by a newline character, meaning the complete list ends with
+             a newline character. */
+            canonicalURL.append("\n");
+        } else {
+            canonicalURL.append("\n");
+        }
+
+        /* Step 1.5 Add the signed headers, followed by a newline character. */
+        strSignedHeader = signedHeaders.substring(0, signedHeaders.length() - 1); // Remove last ";"
+        canonicalURL.append(strSignedHeader).append("\n");
+
+        /* Step 1.6 Use a hash (digest) function like SHA256 to create a hashed value from the payload in the body of
+        the HTTP or HTTPS. */
+        payload = payload == null ? "" : payload;
```

Comment:
shall we change this to a normal if-else statement to make the code more readable.